scripts/lockfile/lockfile.go

```diff
@@ -20,7 +20,9 @@
 
 package main
 
-// This .go file is used to test the lockfile package (m3/src/x/lockfile)
+// This .go file is used to test the lockfile implementation in dbnode/server.
```

Comment:
I wonder if there's a better way for this to work - we're basically forced to export lockfile types from `server` simply because we spin up a new process here, where it would be instead preferable to not export lockfiles from `server` because they're not used anywhere else.

One option would be moving lockfiles into `server/internal/lockfile` (which would also let us keep `Acquire`, `CreateAndAcquire` verbatim) and then have `server/internal/acquire-lock` as a `main` package for spinning up processes, so we keep things unexported outside of an internal package. Thoughts?